### PR TITLE
Analysis refactor gui part5

### PR DESCRIPTION
**Pull request by josenavas**
~~This depends on #2079 so review/merge that one first.~~

Not sure why this is showing so many commits. Anyways, this is adapting the processing page to work both on the study side and the analysis side.

### CODE REVIEW

**Comment by josenavas** [comment]
Fixed the merge conflixt @antgonza thanks for the quick reviews!

File: qiita_db/support_files/patches/python_patches/51.py [hunk 1]
```diff
@@ -445,6 +451,18 @@ def transfer_job(analysis, command_id, params, input_artifact_id, job_data,
                 ['taxa_summary', 'taxa_summary', True, True]]
     TRN.add(sql, sql_args, many=True)
 
+    # Step 8: Give a new client id/client secret pair to the plugins
+    sql = """INSERT INTO qiita.oauth_identifiers (client_id, client_secret)
+                VALUES (%s, %s)"""
+    sr = SystemRandom()
```
Comment (antgonza) [inline qiita_db/support_files/patches/python_patches/51.py]: Could you add some comments about this? What's the idea? Why we need it? etc

File: qiita_db/support_files/patches/python_patches/51.py [hunk 2]
```diff
@@ -445,6 +451,18 @@ def transfer_job(analysis, command_id, params, input_artifact_id, job_data,
...
+    chars = ascii_letters + digits
+    client_id = ''.join(sr.choice(chars) for i in range(50))
```
Comment (antgonza) [inline qiita_db/support_files/patches/python_patches/51.py]: Should all this block be a method?

File: qiita_db/support_files/patches/python_patches/51.py (hunk 1 above)
Comment (josenavas) [inline qiita_db/support_files/patches/python_patches/51.py]: Done

File: qiita_db/support_files/patches/python_patches/51.py (hunk 2 above)
Comment (josenavas) [inline qiita_db/support_files/patches/python_patches/51.py]: Done

File: qiita_db/support_files/patches/python_patches/51.py [hunk 3]
```diff
@@ -491,8 +527,8 @@ def transfer_job(analysis, command_id, params, input_artifact_id, job_data,
         (sum_taxa_cmd_id, 'sort', 'bool', False, 'False'),
         # Beta Diversity
         (bdiv_cmd_id, 'tree', 'string', False, ''),
-        (bdiv_cmd_id, 'metrics',
-         'mchoice:["abund_jaccard","binary_chisq","binary_chord",'
+        (bdiv_cmd_id, 'metric',
```
Comment (antgonza) [inline qiita_db/support_files/patches/python_patches/51.py]: Sorry missed this one in my previous review: Is this change intentional? The actual parameter for that command is: metrics

Comment (josenavas) [inline qiita_db/support_files/patches/python_patches/51.py]: Yes it is intentional, I'm making the command in Qiita to be per-metric based as this will be easier to show the results
